repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -249,12 +249,21 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                     for (String asset : assets) {
                         if(!isAllDomain(asset)) {
                             terms.add(asset);
+
+                            // Add all parent domains in the hierarchy
+                            String currentPath = asset;
+                            while (currentPath.contains("/domain/")) {
+                                allowClauseList.add(mapOf("term", mapOf(QUALIFIED_NAME, currentPath)));
```

Comment:
I think this adds `asset` twice for query?
one in `terms` & another in allowClauseList

Reply:
I suggest adding each qualifiedName in the terms list, it should have same effect without complicating allowClauseList